packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -446,12 +474,14 @@ class InputDecorator extends StatelessWidget {
 
     final TextStyle baseStyle = this.baseStyle ?? themeData.textTheme.subhead;
     final TextStyle hintStyle = decoration.hintStyle ?? baseStyle.copyWith(color: themeData.hintColor);
+    final TextStyle helperStyle =
+      decoration.helperStyle ?? themeData.textTheme.caption.copyWith(color: themeData.hintColor);
     final TextStyle subtextStyle = errorText != null
       ? decoration.errorStyle ?? themeData.textTheme.caption.copyWith(color: themeData.errorColor)
-      : decoration.helperStyle ?? themeData.textTheme.caption.copyWith(color: themeData.hintColor);
+      : helperStyle;
+    final TextStyle counterStyle = decoration.counterStyle ?? helperStyle;
```

Comment:
(trivial nit, feel free to ignore: this block of assignments would be easier to read if each block was line-wrapped the same way)